v3/generator/schemas/schemas.go

```diff
@@ -493,7 +494,7 @@ func IsSimpleSchema(s *base.Schema) bool {
 		return true
 	}
 
-	return s.Type[0] != "object" && s.Type[0] != "map"
+	return s.Type[0] != "object" && s.Type[0] != "map" && s.Type[0] != "array"
```

Comment:
The comment of `IsSimpleSchema` indicates that you explicitly didn't add "array" here because it can't have circular references. It sounds like this function has multiple purposes now.

Why not create a function calles `IsSimpleSchema` so we don't mix the two purposes.
If not we should change the comment of this function.

Reply:
Thanks @sauterp let me check now the usage of this `func` in the codebase and let's see what's fits the best

Reply:
After checking every call, I refactored the comment, and also the usage is better like that, even in the rest of the codebase.
This comment was referencing a logic before I refactored some stuff in the generator